### ISSUE TITLE
Brush Settings Editor numerical sliders shift starting position when switching to different brush properties

### ISSUE THREAD

**Issue by charbelnicolas** [mypaint/mypaint#1129]
#### Description of the problem

The Brush Settings Editor numerical sliders shift their starting position when switching to different brush properties

#### Basic system details

MyPaint version: v2.0.1 Official Appimage
Operating system: Arch Linux
Desktop environment: GNOME 3.38

#### Steps to reproduce

[☞ Tell us how to reproduce the bug in clear, numbered steps.]

1. Start MyPaint
2. Select a brush
3. Click on "Edit Brush Settings"
4. Click on Basic > Eraser)
5. Then click on Basic > Jitter
6. Then click on Basic > Elliptical dab: angle
7. Watch how the numerical sliders move and shift

Please watch video to clearly understand what I am talking about.

[https://youtu.be/6dURP4ygstM](url)

**Comment by jplloyd** [comment]
Thanks for the report, I can confirm that there are some visual issues with the sliders. Should have it fixed by the weekend at the latest.

**Comment by jplloyd** [comment]
Maybe not as bitesize as I thought. The issue comes down to the value text fields having their width based on the number of characters of the max/min value, which varies between brush settings. I have not found a way to fix it through style properties, short of moving the value field above the slider. What's worse is that in the master branch the value field switches position after switching between slider mode and numeric input mode.

I might have to bite the bullet and write a custom widget for sliders with optional numeric input.

**Comment by charbelnicolas** [comment]
That's a shame, thanks for trying, hopefully something can be worked out with a custom widget. By the way, something else I have noticed too while using the numerical sliders is that sometimes when it shows a negative value the "-" minus sign is missing. I'm not sure if this can be worked out too in this same issue or if should create a new one.

**Comment by jplloyd** [comment]
Yes, that was what concerned me, and what I thought was caused by the new slider widget. Turns out it's been an issue for a long time before that was added. I made a fix (bb14904a9dbb02bb730e4e50a90d4f7be173c8e6) for that (could almost certainly be improved) which ensures that the full value is always visible.

The sliders will still change starting positions when switching to a setting with a different number of digits. It appears that the css for that widget node does not respect the min-width attribute. If the general consensus is that having the sliders shift hampers usability, an option could be to use separate labels to set the values (more code, more maintenance).

I'll switch this issue to an enhancement.
